### PR TITLE
Handle dates, not just terms in RightsAssembler

### LINKED ISSUE #113: RightsAssembler date methods need to handle dates, not just terms
## Is your feature request related to a problem? Please describe.
The date methods (`calculate_dates` and `get_date_value`) assume that the start dates and end dates for a given basis or grant are `start_date_period` and `end_date_period`, and cannot currently handle other types of start and end dates.

## Describe the solution you'd like
Either before or in the `calculate_dates` method, check for start and end date types before calling `get_date_value` method.

### PR DESCRIPTION
**Pull request by bonniegee**
Changes to RightsAssembler so that multiple types of dates are added. Updates tests accordingly, including simplifying the check_object_dates method.

Fixes #113

### CODE REVIEW

**Comment by bonniegee** [comment]
Moved this back to draft after realizing tests are failing ~25% of the time. After adding a gazillion print statements, it looks like things go haywire when the start is a date, and the end is a term--working on that fix.

**Comment by bonniegee** [comment]
Noting that this now mostly works, with the (notable) exception of when a term begins 0 years after creation date.

File: assign_rights/assemble.py [hunk 1]
```diff
@@ -72,10 +55,16 @@ def calculate_dates(self, object, request_start_date, request_end_date):
                 representing the group of objects' start and end dates after
                 calculation.
         """
-        object_start = self.get_date_value(
-            object, "start_date", request_start_date, object.start_date_period)
-        object_end = None if object.end_date_open else self.get_date_value(
-            object, "end_date", request_end_date, object.end_date_period)
+        object_start = None
+        object_end = None
+        if getattr(object, "start_date_period"):
```
Comment (helrond) [inline assign_rights/assemble.py]: I would bet this is the offending line. Probably needs to be 
```suggestion
        if getattr(object, "start_date_period") is not None:
```

File: assign_rights/tests.py [hunk 2]
```diff
@@ -193,37 +193,20 @@ def check_object_dates(self, object, request_start_date, request_end_date):
...
-        self.assertEqual(end_date, None)
-
-    def test_calculate_dates(self):
-        """Tests the calculate_dates method."""
+        if object.end_date_open:
+            self.assertEqual(end_date, None)
+        else:
+            self.assertTrue(isinstance(end_date, date))
+
+        if object.start_date_period:
```
Comment (helrond) [inline assign_rights/tests.py:203]: Again, you may need a bit of a different test here:
```suggestion
        if object.start_date_period is not None:
```

File: assign_rights/assemble.py (hunk 1 above)
Comment (bonniegee) [reply assign_rights/assemble.py]: I fixed this another way before seeing this--I flipped the order of which type of start is evaluated first.

Comment (helrond) [reply assign_rights/assemble.py]: OK, that's another way of solving that problem!

**Review by helrond — APPROVED** [review]

**Review by p-galligan — APPROVED** [review]

**Pull request merged**